src/Updater.php

```diff
@@ -371,19 +385,41 @@ protected function downloadPhar()
 
     protected function replacePhar()
     {
-        rename($this->getTempPharFile(), $this->getLocalPharFile());
+        $localPharFile = $this->getLocalPharFile();
+        if (!is_writable($localPharFile)) {
```

Comment:
This code block was moved out from `setLocalPharFile` and you can move it to `getLocalPharFile` method to avoid duplication.